apisix/plugins/ua-restriction.lua

```diff
@@ -58,6 +54,22 @@ local schema = {
             default = "Not allowed"
         },
     },
+    oneOf = {
+        {
+            required = {"allowlist"}
+        },
+        {
+            required = {"denylist"}
+        },
+        {
+            ["not"] = {
+                anyOf = {
+                    {required = {"allowlist"}},
+                    {required = {"denylist"}}
+                }
+            }
+        }
```

Comment:
Why need this?
If both the allowlist and the denylist don't exist, the plugin seems meanless

Reply:
If both allowlist and denylist don't exist, it only checks if the `User-Agent` header exists.
![image](https://github.com/apache/apisix/assets/33349046/0b8f5b81-b733-4366-a4b8-e73b07e0b0ec)

Reply:
No, the `bypass_missing` is designed for the scenario that one of the `allowlist` and `denylist` exists, but the `User-Agent` is missing

Reply:
I'm saying this because of these test cases:
https://github.com/apache/apisix/blob/86004d35d551cddc50d49de23b70f7e62248d68a/t/plugin/ua-restriction.t#L416-L498

I thought it was designed that way. If we don't need this function, I will change it.

Reply:
I don't know why we configure like this, no allowlist and no denylist, it is meanless
```
                        "plugins": {
                            "ua-restriction": {
                            }
                        }
```

Reply:
Fixed.